### PR TITLE
Added selection of oasis restart file if more than one is found.

### LINKED ISSUE #969: Choose oasis restart file if not unambiguously clear
In case of a branchoff simulation: Let the correct oasis restart file be chosen by the user if not unambiguously clear from the content of the restart folder of the parent experiment.

### PR DESCRIPTION
**Pull request by nwieters**
closes #969

### CODE REVIEW

File: src/esm_runscripts/oasis.py
Comment (pgierz) [inline src/esm_runscripts/oasis.py]: Here, we are missing an else. What happens not greater or equal 1 are found (so 0)? Then we should throw an error!

File: src/esm_runscripts/oasis.py
Comment (pgierz) [inline src/esm_runscripts/oasis.py]: This is fine, but at some point (not in this PR, certainly!) we want to move all of these kind of user-info messages into [`loguru.logger`](https://loguru.readthedocs.io/en/stable/).

**Review by pgierz — CHANGES_REQUESTED** [review]
One little "finishing touch" recommendation, but otherwise, looks great! 👍🏻

File: src/esm_runscripts/oasis.py
Comment (nwieters) [reply src/esm_runscripts/oasis.py]: I placed len(branchoff_restart_file) > 1 back again in the else part. Since len(branchoff_restart_file) == 0 is captured by th outer if statement already.

File: src/esm_runscripts/oasis.py
Comment (nwieters) [reply src/esm_runscripts/oasis.py]: Removed

File: src/esm_runscripts/oasis.py
Comment (pgierz) [inline src/esm_runscripts/oasis.py]: Shouldn't this be the other way around?
```suggestion
                        if gconfig["isinteractive"]:
```

File: src/esm_runscripts/oasis.py
Comment (pgierz) [inline src/esm_runscripts/oasis.py]: I'm not entirely sure from memory, but I think you need `.ask()` still here:
```suggestion
                            branchoff_restart_file = os.path.basename(questionary.prompt(questions).ask()["OASIS restart file"])
```

**Review by pgierz — CHANGES_REQUESTED** [review]
Two more little things I saw, have a look 👍🏻

File: src/esm_runscripts/oasis.py
Comment (nwieters) [reply src/esm_runscripts/oasis.py]: I thought so too. But (for a reason I do not understand yet) this will be executed twice, actually four time (for each restart file rstas.nc and rstos.nc) and than a second time which will overwrite everything again. At this second round gconfig["isinteractive"] is False. Thats why I ask here for 'if not'.

**Comment by mandresm** [comment]
I cannot review this right now, please wait for my review before merging (on Monday), I have some comments on the code, that need to be addressed as there are a couple of things (coming from the previous merge) that might result in sneaky errors for the users.

File: src/esm_runscripts/oasis.py
Comment (nwieters) [reply src/esm_runscripts/oasis.py]: This was correct. But I changed it a bit and use questionary.form now. This makes it a few lines shorter.

**Comment by pgierz** [comment]
> I cannot review this right now, please wait for my review before merging (on Monday), I have some comments on the code, that need to be addressed as there are a couple of things (coming from the previous merge) that might result in sneaky errors for the users.

I can make it so that merging requires two approvals, if you want. Since we are a team of three, then everyone has seen everything (unless I forgot how to count), and you could still use "god-mode" to merge for very small changes.

File: src/esm_runscripts/oasis.py
Comment (mandresm) [reply src/esm_runscripts/oasis.py]: It should be `if gconfig["isinteractive"]` because we want to avoid opening questionairs while ESM-Tools runs in the background on a computing node, that will stall the job. If in the second round `gconfig["isinteractive"] is False` as you said @nwieters, then changing this should not be a problem right?

File: src/esm_runscripts/oasis.py
Comment (mandresm) [inline src/esm_runscripts/oasis.py]: @pgierz what's the best Pythonic way to break down this string into multiple lines?

**Review by mandresm — CHANGES_REQUESTED** [review]
I committed some minor changes, and also made a couple of comments. Please, make also sure to change back the `==` in the lresume `if`s of the previous merged PR.

**Comment by mandresm** [comment]
Closes #901 #743 #826

**Comment by mandresm** [comment]
#bump

**Review by mandresm — APPROVED** [review]

**Pull request merged**
